lib/elixir/lib/map_set.ex

```diff
@@ -121,7 +124,7 @@ defmodule MapSet do
       #MapSet<[1, 3]>
 
   """
-  @spec delete(t, value) :: t
+  @spec delete(t(val), val) :: t(val) when val: value
```

Comment:
Sorry @fishcakez, I didn't get what you mean 😕 Do you mean we should have

```elixir
@spec delete(t(val1), val2) :: t(val1) when val1: value, val2: value
```

? Because the second argument is not required to be of the same type as the rest of the values in the set?